mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -0,0 +1,228 @@
+                            .build()));
+            targetCpu = remainingCpu;
+            targetMem = remainingMem;
+          }
+          Protos.TaskInfo task =
+              Protos.TaskInfo
+                  .newBuilder()
+                  .setName("task " + taskId.getValue())
+                  .setTaskId(taskId)
+                  .setSlaveId(offer.getSlaveId())
+                  .addResources(
+                      Protos.Resource.newBuilder().setName("cpus")
+                          .setType(Protos.Value.Type.SCALAR)
+                          .setScalar(Protos.Value.Scalar.newBuilder().setValue(targetCpu)))
+                  .addResources(
+                      Protos.Resource.newBuilder().setName("mem").setType(Protos.Value.Type.SCALAR)
+                          .setScalar(Protos.Value.Scalar.newBuilder().setValue(targetMem)))
+                  .setExecutor(executorBuilder).build();
+
+          launch.addTaskInfos(Protos.TaskInfo.newBuilder(task));
+
+          remainingCpu -= targetCpu;
+          remainingMem -= targetMem;
+        }
+
+        // NOTE: We use the new API `acceptOffers` here to launch tasks.
+        // The 'launchTasks' API will be deprecated.
+        List<Protos.OfferID> offerIds = new ArrayList<Protos.OfferID>();
+        offerIds.add(offer.getId());
+        List<Protos.Offer.Operation> operations = new ArrayList<Protos.Offer.Operation>();
+        Protos.Offer.Operation operation =
+            Protos.Offer.Operation.newBuilder().setType(Protos.Offer.Operation.Type.LAUNCH)
+                .setLaunch(launch).build();
+        operations.add(operation);
+        Protos.Filters filters = Protos.Filters.newBuilder().setRefuseSeconds(1).build();
+        driver.acceptOffers(offerIds, operations, filters);
+      }
+    }
+
+    @Override
+    public void slaveLost(SchedulerDriver driver, Protos.SlaveID slaveId) {
+      System.out.println("Executor " + slaveId.getValue() + " was lost.");
+    }
+
+    @Override
+    public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
+      String taskId = status.getTaskId().getValue();
+      Protos.TaskState state = status.getState();
+      System.out.printf("Task %s is in state %s\n", taskId, state);
+      switch (state) {
+        case TASK_RUNNING:
+        case TASK_FINISHED:
+        case TASK_FAILED:
+        case TASK_KILLED:
+        case TASK_LOST:
+        default:
+      }
+    }
+  }
+
+  private static void usage() {
+    String name = TachyonFramework.class.getName();
+    System.err.println("Usage: " + name + " <hostname>");
+  }
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 1) {
+      usage();
+      System.exit(1);
+    }
+    String hostname = args[0];
+
+    // Start Mesos master. Have Mesos fill in the current user.
+    Protos.FrameworkInfo framework =
+        Protos.FrameworkInfo.newBuilder().setUser("").setName("Test Tachyon Framework")
```

Comment:
- Does pass empty string to `setUser` will make Mesos fill the current user? If yes, state so more explicitly.
- Shall we hard-code the framework name to be "Test Tachyon Framework"?